src/io/directory.js

```diff
@@ -82,4 +89,25 @@ export class Directory extends IOBase {
       });
   }
 
+  getChunkAsBuffer(relativeFilePath, chunkLength) {
+    return this.getPath(relativeFilePath)
+      .then((filePath) => {
+        return new Promise((resolve) => {
+          createReadStream(filePath, {
+            flags: 'r',
+            // This is important because you don't want to encode the
+            // bytes if you are doing a binary check.
+            encoding: null,
+            autoClose: true,
+          })
+          .pipe(
+            firstChunkStream({chunkLength: chunkLength},
+              function(_, enc) {
+                resolve(enc);
+              }
+            )
+          );
+        });
+      });
```

Comment:
I think we want to add a `catch` here to catch any errors inside this code, then a test to make sure the errors bubble up. We do that for most (all?) of our methods that have code inside a `then()` block.